pkg/scheduler/framework/runtime/framework.go

```diff
@@ -829,13 +829,11 @@ func (f *frameworkImpl) RunReservePluginsUnreserve(ctx context.Context, state *f
 }
 
 func (f *frameworkImpl) runReservePluginUnreserve(ctx context.Context, pl framework.ReservePlugin, state *framework.CycleState, pod *v1.Pod, nodeName string) {
-	if !state.ShouldRecordPluginMetrics() {
-		pl.Unreserve(ctx, state, pod, nodeName)
-		return
-	}
 	startTime := time.Now()
 	pl.Unreserve(ctx, state, pod, nodeName)
-	f.metricsRecorder.observePluginDurationAsync(unreserve, pl.Name(), nil, metrics.SinceInSeconds(startTime))
+	if state.ShouldRecordPluginMetrics() {
+		f.metricsRecorder.observePluginDurationAsync(unreserve, pl.Name(), nil, metrics.SinceInSeconds(startTime))
+	}
```

Comment:
only tiny difference, 
- there is a extra check statement while metrics should be recorded, not sure how this impacts the metrics data, ignorable?
- `startTime` become unnecessary when the metrics should not be recorded.

Reply:
- `ShouldRecordPluginMetrics` depend on the variable `recordPluginMetrics`, looks like not last too long.
-  Yes.